server/resp/src/main/java/org/infinispan/server/resp/commands/connection/MemoryEntrySizeUtils.java

```diff
@@ -15,12 +16,30 @@ public static long calculateSize(byte[] key, InternalCacheEntry<byte[], Object>
       return cesc.calculateSize(key, ice);
    }
 
+   public static long calculateSize(Object o) {
+      return sizeof.deepSizeOf(o);
+   }
+
    static long internalCalculateSize(byte[] key, Object value) {
       try {
+         if (value instanceof JsonBucket jb) {
+            // JsonBucket is a special case, we need to use the sizeOf library
+            // to calculate the size of the JsonBucket
+            return sizeof.deepSizeOf(key) + sizeof.deepSizeOf(jb.value())+ objHeader();
+         }
          return sizeof.deepSizeOf(key) + sizeof.deepSizeOf(value);
       } catch (Exception ex) {
          // Try an old style computation
          return pesc.calculateSize(key, value);
       }
    }
+
+   public static long objHeader() {
+      return MemoryEntrySizeUtils.calculateSize(MemoryHeader.EmptyObj);
```

Comment:
Should we just store the value in a static variable so it doesn't need to be calculated every time?

Also I think technically we need to add an additional instance field in the MemoryHeader class to more appropriately estimate the size.

Reply:
Yeah, my initial idea was to provide a general-purpose empty MemoryHeader, since the overhead I've checked that an empty class and a class with a single field have the same overhead.  
However, it seems this isn't a general rule, for i.e. a class with two fields appears to have greater overhead.
I'm adding class specific for JsonBucket